### PR TITLE
Fixes#4184: Add dark mode support to ThirdPartyDependencyListActivity, LicenseListActivity and LicenseTextViewerActivity

### LINKED ISSUE #4184: Add dark mode support to ThirdPartyDependencyListActivity, LicenseListActivity and LicenseTextViewerActivity
Add dark mode support to ThirdPartyDependencyListActivity, LicenseListActivity and LicenseTextViewerActivity

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

### PR DESCRIPTION
**Pull request by aayushimathur6**
## Explanation
Fixes #4184 

Mocks Link:
* https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/0f94055a-c5f7-45a2-8bf5-9dcfdf1c7dce/specs/

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

| First Header  | Second Header |
| ------------- | ------------- | 
|Screenshot_20220514_185423| Screenshot_20220514_185517 |
|Screenshot_20220514_185920|Screenshot_20220514_185834 |
|Screenshot_20220515_201854 | Screenshot_20220515_201650 |
|Screenshot_20220416_191537 | Screenshot_20220416_191251 |
| Screenshot_20220515_201955| Screenshot_20220515_201745 |
| Screenshot_20220423_184652 | Screenshot_20220423_184412 |
| Screenshot_20220424_140509 | Screenshot_20220423_185116 |

### CODE REVIEW

**Comment by aayushimathur6** [comment]
@ayush0402 I have added the dark mode in third party dependency list activity but for license activity, I was unable to do so as I was a bit confused about where to implement it and what colors & also dark mode mocks in wiki don't have images regarding license activity.

**Comment by BenHenning** [comment]
@rt4914 for this PR I think Gradle just sort of awkwardly died in shard 2 (there were no failure logs uploaded, either). I'm just restarting it, though it seems unrelated to the failure in #4268.

**Comment by aayushimathur6** [comment]
@rt4914 PTAL Thanks!

**Comment by BenHenning** [comment]
@aayushimathur6 can you please also add screenshots for the license text view activity? You have some for the list views, but not the one that shows actual license text.

@ayush0402 could you take a first pass review on this?

**Comment by aayushimathur6** [comment]
@BenHenning added the asked PTAL thanks!

**Comment by BenHenning** [comment]
Apologies for the late review.

I'll need to move this back to @rt4914 since his codeowners was restored, and I haven't yet taken a full pass on this PR.

File: app/src/main/res/values/component_colors.xml [hunk 1]
```diff
@@ -36,4 +36,23 @@
...
+  <color name="component_color_license_list_text_viewer_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_license_list_text_viewer_items_activity_background_color">@color/color_palette_dark_text_color</color>
+  <color name="component_color_license_text_viewer_list_options_activity_background_color">@color/color_palette_container_background_color</color>
+  <color name="component_color_license_text_viewer_text_color">@color/color_palette_description_text_color</color>
+
+
+
+
+
+
```
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: Remove these blank lines at the top.

File: app/src/main/res/layout/license_item.xml [hunk 2]
```diff
@@ -14,20 +14,21 @@
...
-      android:background="@drawable/general_item_background_border"
       android:fontFamily="sans-serif-smallcaps"
       android:onClick="@{() -> viewModel.clickOnLicenseItem()}"
       android:paddingStart="16dp"
       android:paddingTop="20dp"
       android:paddingEnd="16dp"
       android:paddingBottom="20dp"
+      android:textColor="@color/component_color_license_list_text_viewer_items_activity_background_color"
       android:text="@{viewModel.licenseName}"
+      android:background="@color/component_color_License_Text_Viewer_List_options_activity_background_color"
```
Comment (rt4914) [inline app/src/main/res/layout/license_item.xml]: Fix the naming to all small_case there should not be any caps character

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aayushimathur6 When I am checking the UI in current app on develop, there are divider lines in between each item. Please have a look and fix them

**Comment by BenHenning** [comment]
Hi. Since I was requested as a reviewer on this PR: Please note that I will be away & unavailable for reviews starting today until 23 May 2022. Please try to make any progress that you can which doesn't require me, and I'll catch up on my reviews when I return. Thanks for your flexibility--I really appreciate it.

**Comment by aayushimathur6** [comment]
@rt4914 completed all the requested changes, PTAL thanks.

File: app/src/main/res/drawable/general_item_background_border_black.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/component_color_third_party_dependency_options_list_activity_background_color" />
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_black.xml]: The naming should be changed to something general so that this file can be used anywhere.

`component_color_general_item_background_solid_color`

File: app/src/main/res/drawable/general_item_background_border_black.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  <stroke
+    android:width="1dp"
+    android:color="@color/component_color_third_party_dependency_list_activity_background_color" />
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_black.xml]: `component_color_general_item_background_stroke_color`

File: app/src/main/res/layout/third_party_dependency_list_fragment.xml [hunk 5]
```diff
@@ -25,9 +25,9 @@
       android:id="@+id/third_party_dependency_list_fragment_recycler_view"
       android:layout_width="match_parent"
       android:layout_height="match_parent"
-      android:background="@color/color_def_white"
+      android:background="@color/component_color_third_party_dependency_list_activity_background_color"
       android:clipToPadding="false"
       android:nestedScrollingEnabled="false"
-      app:list="@{viewModel.thirdPartyDependencyItemList}" />
+      app:list="@{viewModel.thirdPartyDependencyItemList}"/>
```
Comment (rt4914) [inline app/src/main/res/layout/third_party_dependency_list_fragment.xml]: Add ` ` (space) at the end before closing tag `app:list="@{viewModel.thirdPartyDependencyItemList}" />`

File: app/src/main/res/values/component_colors.xml [hunk 6]
```diff
@@ -36,6 +36,19 @@
...
+  <color name="component_color_third_party_dependency_items_text_list_activity_background_color">@color/color_palette_dark_text_color</color>
+  <color name="component_color_third_party_dependency_version_text_view_list_activity_color">@color/color_palette_primary_text_color</color>
+  <!-- License List Activity -->
+  <color name="component_color_license_list_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_license_list_activity_items_background_color">@color/color_palette_container_background_color</color>
+  <!-- License List Text Viewer Activity -->
+  <color name="component_color_license_list_text_viewer_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_license_list_text_viewer_items_activity_background_color">@color/color_palette_dark_text_color</color>
+  <color name="component_color_license_text_viewer_list_options_activity_background_color">@color/color_palette_container_background_color</color>
+  <color name="component_color_license_text_viewer_text_color">@color/color_palette_description_text_color</color>
```
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: Naming of all components is sort of incorrect. Follow this format

`component_color_<activity-name>_<type-of-color>_color`

Example

`component_color_third_party_dependency_list_activity_background_color ` -> this is correct
`component_color_third_party_dependency_version_text_view_list_activity_color ` should be renamed to `component_color_third_party_dependency_list_activity_version_text_color `

Basically all internal layouts like `fragments`, `items` will also use `activity` name only.

Example `component_color_license_list_text_viewer_activity_background_color ` -> `component_color_license_text_viewer_activity_background_color `

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aayushimathur6 Suggested changes.

Thanks.

**Comment by bhaktideshmukh** [comment]
@aayushimathur6 Are the screenshots recently updated?

**Comment by aayushimathur6** [comment]
@bhaktideshmukh yes I have uploaded the screenshot after the updation for the divider line which wasn't visible earlier but now it is and also done with all the changes asked.

**Review by Akshatkamboj14 — APPROVED** [review]
LGTM, thanks.

**Comment by aayushimathur6** [comment]
> @aayushimathur6 When I am checking the UI in current app on develop, there are divider lines in between each item. Please have a look and fix them

done

**Comment by aayushimathur6** [comment]
> @aayushimathur6 When I am checking the UI in current app on develop, there are divider lines in between each item. Please have a look and fix them

done

**Review by bhaktideshmukh — APPROVED** [review]
LGTM, Thanks.

File: app/src/main/res/drawable/general_item_background_border_black.xml (hunk 3 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_black.xml]: done.

File: app/src/main/res/drawable/general_item_background_border_black.xml (hunk 4 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_black.xml]: done

File: app/src/main/res/layout/third_party_dependency_list_fragment.xml (hunk 5 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/third_party_dependency_list_fragment.xml]: done

File: app/src/main/res/values/component_colors.xml (hunk 6 above)
Comment (aayushimathur6) [reply app/src/main/res/values/component_colors.xml]: done

File: app/src/main/res/layout/license_item.xml [hunk 7]
```diff
@@ -14,19 +14,20 @@
...
       android:layout_width="match_parent"
-      android:background="@drawable/general_item_background_border"
+      android:background="@drawable/general_item_background_border_black"
       android:fontFamily="sans-serif-smallcaps"
       android:onClick="@{() -> viewModel.clickOnLicenseItem()}"
       android:paddingStart="16dp"
       android:paddingTop="20dp"
       android:paddingEnd="16dp"
       android:paddingBottom="20dp"
+      android:textColor="@color/component_color_license_list_text_viewer_items_activity_background_color"
```
Comment (rt4914) [inline app/src/main/res/layout/license_item.xml]: This should be renamed to 
`component_color_license_list_activity_license_name_text_view_color `

There were couple of mistakes with earlier name

1. It's not a background color.
2. The activity name was not used correctly.

File: app/src/main/res/layout/license_text_viewer_fragment.xml [hunk 8]
```diff
@@ -11,15 +11,17 @@
   <ScrollView
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@color/color_def_white">
+    android:background="@color/component_color_license_list_text_viewer_activity_background_color">
```
Comment (rt4914) [inline app/src/main/res/layout/license_text_viewer_fragment.xml]: There is no `LicenseListTextViewActivity` and therefore this name is incorrect. I think it should be
`component_color_license_text_viewer_activity_background_color `

File: app/src/main/res/layout/license_text_viewer_fragment.xml [hunk 9]
```diff
@@ -11,15 +11,17 @@
...
 
     <TextView
       android:id="@+id/copyright_license_text_view"
       android:layout_width="match_parent"
       android:layout_height="wrap_content"
+      android:background="@drawable/general_item_background_border_black"
       android:layout_gravity="center_horizontal"
       android:autoLink="web"
       android:padding="32dp"
+      android:textColor="@color/component_color_license_list_text_viewer_activity_text_color"
```
Comment (rt4914) [inline app/src/main/res/layout/license_text_viewer_fragment.xml]: Ditto. Incorrect name

File: app/src/main/res/layout/third_party_dependency_item.xml [hunk 10]
```diff
@@ -34,7 +33,8 @@
         android:paddingTop="16dp"
         android:paddingEnd="16dp"
         android:paddingBottom="4dp"
-        android:text="@{viewModel.dependencyName}" />
+        android:textColor="@color/component_color_third_party_dependency_list_activity_items_background_color"
```
Comment (rt4914) [inline app/src/main/res/layout/third_party_dependency_item.xml]: `component_color_third_party_dependency_list_activity_dependency_name_text_view_color `

Its not `items` , instead its a proper textview.
Also, not background but text color.

File: app/src/main/res/layout/third_party_dependency_item.xml [hunk 11]
```diff
@@ -34,7 +33,8 @@
...
+        android:text="@{viewModel.dependencyName}"/>
```
Comment (rt4914) [inline app/src/main/res/layout/third_party_dependency_item.xml]: Spacing issue
`android:text="@{viewModel.dependencyName}" />`

File: app/src/main/res/layout/third_party_dependency_item.xml [hunk 12]
```diff
@@ -45,9 +45,9 @@
         android:paddingTop="4dp"
         android:paddingEnd="16dp"
         android:paddingBottom="16dp"
-        android:textColor="@color/color_def_accessible_grey"
         android:text="@{viewModel.dependencyVersion}"
-        android:textSize="12sp" />
+        android:textColor="@color/component_color_third_party_dependency_list_activity_version_text_color"
+        android:textSize="12sp"/>
```
Comment (rt4914) [inline app/src/main/res/layout/third_party_dependency_item.xml]: Spacing issue
`android:textSize="12sp" />`

File: app/src/main/res/layout/third_party_dependency_item.xml [hunk 13]
```diff
@@ -45,9 +45,9 @@
         android:paddingTop="4dp"
         android:paddingEnd="16dp"
         android:paddingBottom="16dp"
-        android:textColor="@color/color_def_accessible_grey"
         android:text="@{viewModel.dependencyVersion}"
-        android:textSize="12sp" />
+        android:textColor="@color/component_color_third_party_dependency_list_activity_version_text_color"
```
Comment (rt4914) [inline app/src/main/res/layout/third_party_dependency_item.xml]: `component_color_third_party_dependency_list_activity_dependency_version_text_color `

File: app/src/main/res/values/component_colors.xml [hunk 14]
```diff
@@ -36,6 +38,19 @@
...
+  <!-- Third Party Dependency List Activity -->
+  <color name="component_color_third_party_dependency_list_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_third_party_dependency_list_activity_options_background_color">@color/color_palette_container_background_color</color>
+  <color name="component_color_third_party_dependency_list_activity_items_background_color">@color/color_palette_dark_text_color</color>
+  <color name="component_color_third_party_dependency_list_activity_version_text_color">@color/color_palette_primary_text_color</color>
+  <!-- License List Activity -->
+  <color name="component_color_license_list_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_license_list_activity_items_background_color">@color/color_palette_container_background_color</color>
+  <!-- License List Text Viewer Activity -->
+  <color name="component_color_license_list_text_viewer_activity_background_color">@color/color_palette_dark_background_color</color>
```
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: For all values under this header the name should start witth
`component_color_license_list_text_viewer_activity_...` which is not the case for all .

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aayushimathur6 PTAL Thanks.

File: app/src/main/res/layout/license_item.xml (hunk 7 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/license_item.xml]: done

File: app/src/main/res/layout/license_text_viewer_fragment.xml (hunk 8 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/license_text_viewer_fragment.xml]: done

File: app/src/main/res/layout/license_text_viewer_fragment.xml (hunk 9 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/license_text_viewer_fragment.xml]: corrected

File: app/src/main/res/layout/third_party_dependency_item.xml (hunk 10 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/third_party_dependency_item.xml]: done

File: app/src/main/res/layout/third_party_dependency_item.xml (hunk 11 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/third_party_dependency_item.xml]: updated

File: app/src/main/res/layout/third_party_dependency_item.xml (hunk 12 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/third_party_dependency_item.xml]: updated

File: app/src/main/res/layout/third_party_dependency_item.xml (hunk 13 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/third_party_dependency_item.xml]: done

File: app/src/main/res/values/component_colors.xml (hunk 14 above)
Comment (aayushimathur6) [reply app/src/main/res/values/component_colors.xml]: corrected

**Comment by aayushimathur6** [comment]
@rt4914 updated all the asked changes, PTAL

File: app/src/main/res/layout/third_party_dependency_item.xml [hunk 15]
```diff
@@ -19,12 +19,11 @@
     <LinearLayout
       android:layout_width="match_parent"
       android:layout_height="wrap_content"
-      android:background="@drawable/general_item_background_border"
+      android:background="@drawable/general_item_background_border_black"
       android:onClick="@{(v) -> viewModel.clickOnThirdPartyDependencyItem()}"
       android:orientation="vertical"
       app:layout_constraintStart_toStartOf="parent"
       app:layout_constraintTop_toTopOf="parent">
-
```
Comment (rt4914) [inline app/src/main/res/layout/third_party_dependency_item.xml]: Add this blank line back.

Undo this change.

**Review by rt4914 — APPROVED** [review]
@aayushimathur6 One nit change, otherwise LGTM

File: app/src/main/res/layout/third_party_dependency_item.xml (hunk 15 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/third_party_dependency_item.xml]: done

**Comment by aayushimathur6** [comment]
@rt4914 done with last change
